packages/react-native/sdks/hermes-engine/hermes-utils.rb

```diff
@@ -177,8 +178,54 @@ def podspec_source_build_from_github_tag(react_native_path)
 end
 
 def podspec_source_build_from_github_main()
-    hermes_log("Using the latest commit from main.")
-    return {:git => HERMES_GITHUB_URL, :commit => `git ls-remote #{HERMES_GITHUB_URL} main | cut -f 1`.strip}
+    # [macOS
```

Comment:
Rather than remove the lines, can we comment these out, and then move all the merge-base logic to a separate named method (As you had it before) so that this particular diff is easier to read? Something like
```
    # hermes_log("Using the latest commit from main.")
    # return {:git => HERMES_GITHUB_URL, :commit => `git ls-remote #{HERMES_GITHUB_URL} main | cut -f 1`.strip}
  # [macOS This logic is different....
   commit = getCommitForMergeBase()
   hermes_log("Using Hermes commit hash from the merge base with React Native main: #{commit}")
   return {:git => HERMES_GITHUB_URL, :commit => commit}
  # macOS]


```

Reply:
Done.